src/hotspot/share/opto/superword.hpp

```diff
@@ -657,9 +657,16 @@ class SWPointer : public ArenaObj {
   bool        _analyze_only; // Used in loop unrolling only for swpointer trace
   uint        _stack_idx;    // Used in loop unrolling only for swpointer trace
 
-  PhaseIdealLoop* phase() const { return _slp->phase(); }
-  IdealLoopTree*  lpt() const   { return _slp->lpt(); }
-  PhiNode*        iv() const    { return _slp->iv();  } // Induction var
+  PhaseIdealLoop* _phase;    // PhaseIdealLoop handle
+  IdealLoopTree*  _lpt;      // Current IdealLoopTree
+
+  PhaseIdealLoop* phase() const { return _phase; }
+  IdealLoopTree*  lpt() const   { return _lpt; }
+  PhiNode* iv() const {
+    return _slp ? _slp->iv() : _lpt->_head->as_CountedLoop()->phi()->as_Phi();
+  }
+
+  void init();
```

Comment:
This is just a helper function for the constructors, right? Maybe move it closer to them?

Reply:
Done